core/domain/suggestion_services.py

```diff
@@ -1826,3 +2020,866 @@ def update_question_suggestion(
+        ) += int(suggestion.edited_by_reviewer)
+        _update_question_contribution_stats_models(
+            [question_contribution_stat])
+
+
+def update_question_review_stats(
+    suggestion: suggestion_registry.BaseSuggestion
+) -> None:
+    """Creates/updates QuestionReviewStatsModel model for given question
+    reviewer when a question is reviewed.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            reviewed.
+    """
+    topic_ids = [
+        topic.topic_id
+        for topic in skill_services.get_all_topic_assignments_for_skill(
+            suggestion.target_id)
+    ]
+    suggestion_is_accepted = (
+        suggestion.status == suggestion_models.STATUS_ACCEPTED
+    )
+
+    for topic_id in topic_ids:
+        question_review_stat_model = (
+            suggestion_models.QuestionReviewStatsModel.get(
+                str(suggestion.final_reviewer_id), topic_id
+            ))
+
+        if question_review_stat_model is None:
+            suggestion_models.QuestionReviewStatsModel.create(
+                reviewer_user_id=str(suggestion.final_reviewer_id),
+                topic_id=topic_id,
+                reviewed_questions_count=1,
+                accepted_questions_count=(1 if suggestion_is_accepted else 0),
+                accepted_questions_with_reviewer_edits_count=int(
+                    suggestion.edited_by_reviewer),
+                first_contribution_date=_get_date_as_string(
+                    suggestion.last_updated),
+                last_contribution_date=_get_date_as_string(
+                    suggestion.last_updated)
+            )
+            continue
+
+        question_review_stat = (
+            _create_question_review_stats_from_model(
+                question_review_stat_model))
+
+        increment_question_review_stats(
+            question_review_stat, suggestion.last_updated,
+            suggestion_is_accepted,
+            suggestion.edited_by_reviewer)
+        _update_question_review_stats_models([question_review_stat])
+
+    update_question_contribution_stats_at_review(suggestion)
+
+
+def increment_translation_contribution_stats_at_review(
+    translation_contribution_stat: (
+        suggestion_registry.TranslationContributionStats),
+    content_word_count: int,
+    suggestion_is_accepted: bool,
+    edited_by_reviewer: bool
+) -> None:
+    """Updates TranslationContributionStats object.
+
+    Args:
+        translation_contribution_stat: TranslationContributionStats. The stats
+            object to update.
+        content_word_count: int. The number of words in the translation.
+        suggestion_is_accepted: bool. A flag that indicates whether the
+            suggestion is accepted.
+        edited_by_reviewer: bool. A flag that indicates whether the suggestion
+            is edited by the reviewer.
+    """
+    translation_contribution_stat.accepted_translations_count += int(
```

Comment:
Please fix this, let's not cast. Ditto elsewhere.

Reply:
Done